### PR TITLE
Add Propose script

### LINKED ISSUE #2: Create Scripts
- [x] Deployment Script — For deploying a candidate upgrade Governor
- [x] Proposal Script — For a PT delegate with sufficient voting weight to run to add the proposal to PT's Governance Queue

### PR DESCRIPTION
**Pull request by alexkeating**
#### Description
- Add propose script
- Add GovernorAlpha interface

Closes #2

### CODE REVIEW

**Comment by alexkeating** [comment]
Current dependencies on/for this PR:
* main
  * **PR #6** <a href="https://app.graphite.dev/github/pr/ScopeLift/pooltogether-governor-upgrade/6" target="_blank">Graphite</a> 
    * **PR #7** <a href="https://app.graphite.dev/github/pr/ScopeLift/pooltogether-governor-upgrade/7" target="_blank">Graphite</a>  👈
      * **PR #9** <a href="https://app.graphite.dev/github/pr/ScopeLift/pooltogether-governor-upgrade/9" target="_blank">Graphite</a> 
        * **PR #11** <a href="https://app.graphite.dev/github/pr/ScopeLift/pooltogether-governor-upgrade/11" target="_blank">Graphite</a> 
          * **PR #12** <a href="https://app.graphite.dev/github/pr/ScopeLift/pooltogether-governor-upgrade/12" target="_blank">Graphite</a> 
            * **PR #13** <a href="https://app.graphite.dev/github/pr/ScopeLift/pooltogether-governor-upgrade/13" target="_blank">Graphite</a> 
              * **PR #14** <a href="https://app.graphite.dev/github/pr/ScopeLift/pooltogether-governor-upgrade/14" target="_blank">Graphite</a> 
                * **PR #15** <a href="https://app.graphite.dev/github/pr/ScopeLift/pooltogether-governor-upgrade/15" target="_blank">Graphite</a> 
                  * **PR #17** <a href="https://app.graphite.dev/github/pr/ScopeLift/pooltogether-governor-upgrade/17" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/ScopeLift/pooltogether-governor-upgrade/7?utm_source=stack-comment).

File: src/interfaces/IGovernorAlpha.sol [hunk 1]
```diff
@@ -0,0 +1,75 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.18;
+
```
Comment (alexkeating) [inline src/interfaces/IGovernorAlpha.sol:3]: *(arbitray place)*

- This interface is the same as gitcoin sans the `gtc` method
- I also verified this matches the Pooltogether governor alpha

File: script/Propose.s.sol [hunk 2]
```diff
@@ -0,0 +1,49 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.18;
```
Comment (alexkeating) [inline script/Propose.s.sol:2]: *(arbitrary comment)*

- This is the same as gitcoin except I updated the addresses and proposer comment.

File: script/Propose.s.sol [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+import {Script} from "forge-std/Script.sol";
+import {ICompoundTimelock} from
+  "@openzeppelin/contracts/governance/extensions/GovernorTimelockCompound.sol";
+
+import {PooltogetherGovernor} from "src/PooltogetherGovernor.sol";
+import {IGovernorAlpha} from "src/interfaces/IGovernorAlpha.sol";
+
+contract Propose is Script {
```
Comment (apbendi) [inline script/Propose.s.sol:11]: Similar to previous PR, let's update the README to include instructions for using and testing this script

File: script/Propose.s.sol [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    return GOVERNOR_ALPHA.propose(
+      _targets, _values, _signatures, _calldatas, "Upgrade to Governor Bravo"
+    );
+  }
+
+  /// @dev After the new Governor is deployed on mainnet, this can move from a parameter to a const
+  function run(PooltogetherGovernor _newGovernor) public returns (uint256 _proposalId) {
+    // The expectation is the key loaded here corresponds to the address of the `proposer` above.
+    // When running as a script, broadcast will fail if the key is not correct.
+    uint256 _proposerKey = vm.envUint("PROPOSER_PRIVATE_KEY");
```
Comment (apbendi) [inline script/Propose.s.sol:41]: Should be added to the .env template file

**Review by apbendi — CHANGES_REQUESTED** [review]
Small stuff, otherwise looking good. Tested locally and validated it worked after changing the proposer address—or at least that I can get to the revert condition of `GovernorAlpha::propose: proposer votes below proposal threshold` as expected 👍

File: script/Propose.s.sol (hunk 3 above)
Comment (alexkeating) [reply script/Propose.s.sol:11]: https://github.com/ScopeLift/pooltogether-governor-upgrade/pull/7/commits/e361ba75debff3955ebd0f3ab5c0adac08031ac6

File: script/Propose.s.sol (hunk 4 above)
Comment (alexkeating) [reply script/Propose.s.sol:41]: I just added this to the previous pr #6

**Pull request merged**